Source/SourceKittenFramework/SourceKitVariant.swift

```diff
@@ -0,0 +1,447 @@
+        get { return box.array }
+        set { box.array = newValue }
+    }
+
+    public var dictionary: [String: SourceKitVariant]? {
+        get { return box.dictionary }
+        set { box.dictionary = newValue }
+    }
+
+    public var string: String? {
+        get { return box.string }
+        set { box.string = newValue }
+    }
+
+    public var int: Int? {
+        get { return box.int }
+        set { box.int = newValue }
+    }
+
+    public var bool: Bool? {
+        get { return box.bool }
+        set { box.bool = newValue }
+    }
+
+    public var any: Any? { return box.any }
+
+    public subscript(string: String) -> SourceKitVariant? {
+        get { return box.dictionary?[string] }
+        set { box.dictionary?[string] = newValue }
+    }
+
+    public subscript(key: SwiftDocKey) -> SourceKitVariant? {
+        get { return box.dictionary?[key.rawValue] }
+        set { box.dictionary?[key.rawValue] = newValue }
+    }
+
+    internal init(variant: sourcekitd_variant_t, response: sourcekitd_response_t) {
+        box = _VariantBox(variant: .variant(variant, _ResponseBox(response)))
+    }
+
+    fileprivate let box: _VariantBox
+}
+
+// MARK: - Convenient properties for SwiftDocKey
+extension SourceKitVariant {
+    /// Annotated declaration (String).
+    public var annotatedDeclaration: String?
+        { return self[.annotatedDeclaration]?.string }
+    /// Body length (Int).
+    public var bodyLength: Int?
+        { return self[.bodyLength]?.int }
+    /// Body offset (Int).
+    public var bodyOffset: Int? { return self[.bodyOffset]?.int }
+    /// Diagnostic stage (String).
+    public var diagnosticStage: String? { return self[.diagnosticStage]?.string }
+    /// File path (String).
+    public var filePath: String? { return self[.filePath]?.string }
+    /// Full XML docs (String).
+    public var fullXMLDocs: String? { return self[.fullXMLDocs]?.string }
+    /// Kind (SourceKitVariant.string).
+    public var kind: SourceKitVariant? { return self[.kind] }
+    /// Length (Int).
+    public var length: Int? { return self[.length]?.int }
+    /// Name (String).
+    public var name: String? { return self[.name]?.string }
+    /// Name length (Int).
+    public var nameLength: Int? { return self[.nameLength]?.int }
+    /// Name offset (Int).
+    public var nameOffset: Int? { return self[.nameOffset]?.int }
+    /// Offset (Int).
+    public var offset: Int? { return self[.offset]?.int }
+    /// Substructure ([SourceKitVariant]).
+    public var substructure: [SourceKitVariant]? { return self[.substructure]?.array }
+    /// Syntax map ([SourceKitVariant]).
+    public var syntaxMap: [SourceKitVariant]? { return self[.syntaxMap]?.array }
+    /// Type name (String).
+    public var typeName: String? { return self[.typeName]?.string }
+    /// Inheritedtype ([SourceKitVariant])
+    public var inheritedtypes: [SourceKitVariant]? { return self[.inheritedtypes]?.array }
+}
+
+// MARK: - Accessors for SwiftDocKey
```

Comment:
Ideally we'd just remove `SwiftDocKey`. Or at least just make it a private implementation detail.